### ISSUE TITLE
Successful integration with angular-cli project

### ISSUE THREAD

**Issue by arproudlock** [dougludlow/ng2-bs3-modal#31]
Apologies in advance.  This is not an issue, wasn't sure where to put this, but might be helpful for some:

**Step 1: Install**
`npm install --save ng2-bs3-modal`

**Step 2: Include in Vendor files**
Open ember-cli-build.js
include 'ng2-bs3-modal' in vendorNpmFiles array:

```
var Angular2App = require('angular-cli/lib/broccoli/angular2-app');

module.exports = function(defaults) {
  var app = new Angular2App(defaults, {
    vendorNpmFiles: [
      'angularfire2/**/*.js',
      'firebase/lib/*.js',
      'ng2-bs3-modal/**/*.js'
    ]
  });
  return app.toTree();
};
```

`Step 3: Build`
`ng build`
Check `/dist/vendor` for ng2-bs3-modal folder

**Step 4: Include Scripts and CSS in index.html**
Open index.html
In the head include:
`<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.6/css/bootstrap.css">`

In the body (with the other script src tags) include:
`  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.0/jquery.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.6/js/bootstrap.js"></script>`

**Step 5: System.js**
In index.html, in `System.config({})` add 'ng2-bs3-modal' to packages and map.

Example:

```
System.config({
      packages: {
        app: {
          format: 'register',
          defaultExtension: 'js'
        },
        'ng2-material': {
            format: 'cjs',
            defaultExtension: 'js'
        },
        angularfire2: {
            defaultExtension: 'js',
            main: 'angularfire2.js'
        },
        'ng2-bs3-modal': {
            defaultExtension: 'js',
            main: 'ng2-bs3-modal.js'
        }
      },
      map: {
        firebase: 'vendor/firebase/lib/firebase-web.js',
        angularfire2: 'vendor/angularfire2',
        'ng2-bs3-modal': 'vendor/ng2-bs3-modal'
      }
    });
System.import('app.js').then(null, console.error.bind(console));
```

Step 6: Import in component ts file and include in directives array

`import { MODAL_DIRECTIVES} from 'ng2-bs3-modal/ng2-bs3-modal';`

`directives: [ROUTER_DIRECTIVES, CORE_DIRECTIVES, MATERIAL_DIRECTIVES, MODAL_DIRECTIVES],`

Should now be able to use ng2-bs3-modal with you angular-cli project.

**Comment by dougludlow** [comment]
:thumbsup: 

Thanks @arproudlock, I've added a link from the readme to this issue for an example for angular-cli.

**Comment by haneyman** [comment]
This was helpful but it looks like things have changed with the CLI.  I am using version 1.0.0-beta.1:  
**Step 2** should now be: In file angular-cli-build.js, add 'ng2-bs3-modal/**/*.js' 
**Step 5**. should now be: In file system-config.ts, add 'ng2-bs3-modal' to the barrels array, add 'vendor/ng2-bs3-modal' to the system.config map

**Comment by Toadskin** [comment]
I'm trying to adhere to the "do not use jQuery" recommendation so will look around for a "pure" Angular 2.0 modal if one is available.

**Comment by sudhashri** [comment]
I have followed all the sets mentioned in Post 1 and 3 but I get the following error.
`zone.js:101 GET http://localhost:4200/vendor/ng2-bs3-modal/ng2-bs3-modal 404 (Not Found)`

Here are my files:

1) `package.json`
```
......
"dependencies": {
    "@angular/common": "2.0.0-rc.4",
    "@angular/compiler": "2.0.0-rc.4",
    "@angular/core": "2.0.0-rc.4",
    "@angular/forms": "0.2.0",
    "@angular/http": "2.0.0-rc.4",
    "@angular/platform-browser": "2.0.0-rc.4",
    "@angular/platform-browser-dynamic": "2.0.0-rc.4",
    "@angular/router": "3.0.0-beta.2",
    "bootstrap": "^3.3.7",
    "es6-shim": "0.35.1",
    "jspdf": "^1.2.61",
    "ng2-bs3-modal": "^0.7.0",
    "reflect-metadata": "0.1.3",
    "rxjs": "5.0.0-beta.6",
    "systemjs": "0.19.26",
    "zone.js": "0.6.12"
  }
....
```
2) `angular-cli-build.js`
```
var Angular2App = require('angular-cli/lib/broccoli/angular2-app');

module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
      'systemjs/dist/system-polyfills.js',
      'systemjs/dist/system.src.js',
      'zone.js/dist/**/*.+(js|js.map)',
      'es6-shim/es6-shim.js',
      'reflect-metadata/**/*.+(ts|js|js.map)',
      'rxjs/**/*.+(js|js.map)',
      '@angular/**/*.+(js|js.map)',
      'jspdf/dist/jspdf.min.js',
      'ng2-bs3-modal/**/*.js'
    ]
  });
};
```
3) `system-config.ts`
```
"use strict";

// SystemJS configuration file, see links for more information
// https://github.com/systemjs/systemjs
// https://github.com/systemjs/systemjs/blob/master/docs/config-api.md

/***********************************************************************************************
 * User Configuration.
 **********************************************************************************************/
/** Map relative paths to URLs. */
const map: any = {
  'jspdf': 'vendor/jspdf/dist/jspdf.min.js'
};

/** User packages configuration. */
const packages: any = {
  'jspdf': {format: 'global'}
};

////////////////////////////////////////////////////////////////////////////////////////////////
/***********************************************************************************************
 * Everything underneath this line is managed by the CLI.
 **********************************************************************************************/
const barrels: string[] = [
  // Angular specific barrels.
  '@angular/core',
  '@angular/common',
  '@angular/compiler',
  '@angular/forms',
  '@angular/http',
  '@angular/router',
  '@angular/platform-browser',
  '@angular/platform-browser-dynamic',

  // Thirdparty barrels.
  'rxjs',

  // App specific barrels.
  'app',
  'app/shared',
  /** @cli-barrel */
];

const cliSystemConfigPackages: any = {};
barrels.forEach((barrelName: string) => {
  cliSystemConfigPackages[barrelName] = { main: 'index' };
});

/** Type declaration for ambient System. */
declare var System: any;

// Apply the CLI SystemJS configuration.
System.config({
  map: {
    '@angular': 'vendor/@angular',
    'rxjs': 'vendor/rxjs',
    'main': 'main.js',
    'ng2-bs3-modal': 'vendor/ng2-bs3-modal'
  },
  packages: cliSystemConfigPackages
});

// Apply the user's configuration.
System.config({ map, packages });
```
4) `index.html`
```
<!doctype html>
<html>
<head>
  <meta charset="utf-8">
  <title>ModalTest</title>
  <base href="/">

  {{#unless environment.production}}
  <script src="/ember-cli-live-reload.js" type="text/javascript"></script>
  {{/unless}}
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="icon" type="image/x-icon" href="favicon.ico">
  <link rel="stylesheet" href="app/styles.css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.6/css/bootstrap.css">
  <!--  For jsPDF and jsPDF-autotable -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jspdf/1.0.272/jspdf.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jspdf-autotable/2.0.16/jspdf.plugin.autotable.js"></script>
</head>
<body>
  <app-root>Loading...</app-root>

    {{#each scripts.polyfills}}
    <script src="{{.}}"></script>
    {{/each}}
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.0/jquery.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.6/js/bootstrap.js"></script>
    <script>
      System.import('system-config.js').then(function () {
        System.import('main');
      }).catch(console.error.bind(console));
    </script>

</body>
</html>
```
5) `app.component.ts`
```
import { Component } from '@angular/core';
import { MODAL_DIRECTIVES} from 'ng2-bs3-modal/ng2-bs3-modal';
declare let jsPDF;
@Component({
  moduleId: module.id,
  selector: 'app-root',
  templateUrl: 'app.component.html',
  styleUrls: ['app.component.css'],
  directives: [MODAL_DIRECTIVES],
})
export class AppComponent {
  title = 'app works!';
  private item = {
      "Name" : "XYZ",
      "Age" : "22",
      "Gender" : "Male"
  }
  createFile(){
      var doc = new jsPDF();
      var i=0;
      for(var key in this.item){
          doc.text(20, 10 + i, key + ": " + this.item[key]);
          i+=10;
      }
      doc.save('Test.pdf');

      // Send the document
      return doc;
  }
}
```
6) `app.component.html`
```
<button type="button" class="btn btn-default" (click)="modal.open()">Open me!</button>

<modal #modal>
    <modal-header [show-close]="true">
        <h4 class="modal-title">I'm a modal!</h4>
    </modal-header>
    <modal-body>
        Hello World!
    </modal-body>
    <modal-footer [show-default-buttons]="true"></modal-footer>
</modal>
```
7) `main.ts`
```
import { bootstrap } from '@angular/platform-browser-dynamic';
import { enableProdMode } from '@angular/core';
import { AppComponent, environment } from './app/';

if (environment.production) {
  enableProdMode();
}

bootstrap(AppComponent);
```

Should I make any changes to my files to make Modal work?

**Comment by madhurahuja** [comment]
Hi Sudhashri, you can get rid of this error by using 
 `defaultJSExtensions: true`
Put this in System.config.

**Comment by sudhashri** [comment]
Adding `defaultJSExtensions: true` did the trick for me. Thank you @madhurahuja

**Comment by sauverma16** [comment]
I am not able to make this work still. Here are the steps I did.
1. ng new trial
2. installed bootstrap3 and ng2-bs3-modal
3. Added  'ng2-bs3-modal/**/*.js' to vendorNPM files in angular-cli-build.js
4. ng build and check dist folder - All OK
5. Followed step 4 except changing bootstrap version to 3.3.7 : All OK 
6. Step 5:  In system-config.ts. Added the 'ng2-bs3-modal': 'vendor/ng2-bs3-modal' to map
and also   defaultJSExtensions: true,
7. In my In system-config.ts packages is set to the following. 
packages: cliSystemConfigPackages

The moment I try to tweak packages to add 
        'ng2-bs3-modal': {
            defaultExtension: 'js',
            main: 'ng2-bs3-modal.js'
        }

I start getting this error : zone.js:101 GET http://localhost:4200/vendor/@angular/platform-browser-dynamic.js 404 (Not Found)

8. Alternatively, If I keep the packages: cliSystemConfigPackages intact and add the following directive to my component
  directives: [MODAL_DIRECTIVES],

I start getting this error:
zone.js:101 GET http://localhost:4200/vendor/ng2-bs3-modal/ng2-bs3-modal.js 404 (Not Found)

Pl help. I have tried a clean install and tweaking many things but cannot get this to work. Can someone will more expertise help everyone by cleaning up the instructions.
